lib/cadet/jobs/updater/CS1101S.ex

```diff
@@ -42,34 +42,25 @@ defmodule Cadet.Updater.CS1101S do
     git("pull", ["origin", "master"])
   end
 
-  defp git("clone", args) do
-    {out, exit} =
-      System.cmd(
-        "git",
-        ["clone"] ++ args,
-        env: [{"GIT_SSH_COMMAND", "ssh -i #{@key_file}"}],
-        stderr_to_stdout: true
-      )
-
-    Logger.debug(fn ->
-      "git clone exited with #{exit}: #{out}"
-    end)
-
-    {out, exit}
-  end
-
   defp git(cmd, args) do
+    extra_args = if cmd == "clone", do: [], else: ["-C", @local_name]
+
     {out, exit} =
       System.cmd(
         "git",
-        ["-C", @local_name] ++ [cmd] ++ args,
+        extra_args ++ [cmd] ++ args,
         env: [{"GIT_SSH_COMMAND", "ssh -i #{@key_file}"}],
         stderr_to_stdout: true
       )
 
-    Logger.debug(fn ->
-      "git #{cmd} #{inspect(args)} exited with #{exit}: #{out}"
-    end)
+    if exit != 0 do
+      error_message = "git #{cmd} #{inspect(args)} errored with exit code #{exit}\n#{out}"
+      Logger.error(error_message)
+      Sentry.capture_message(error_message)
+    else
+      Logger.info("git #{cmd} #{inspect(args)} succeeded")
+      Logger.debug(fn -> "\n#{out}" end)
```

Comment:
I don't quite understand why this is wrapped in a function

Reply:
Oh, it's from credo basically it's recommended for `Logger.debug`'s argument to be evaluated lazily because it might contain something expensive, and sometimes in the config, `:logger` is set to not emit `debug` logs.

Reply:
Oh otherwise it won't be lazily evaluated even if the log level is set to not report it. Thanks for the explanation!

Reply:
Elixir 1.7 fixes this: https://elixir-lang.org/blog/2018/07/25/elixir-v1-7-0-released/#logger-compile-time-purging